kroxylicious-filters/pom.xml

```diff
@@ -16,15 +16,16 @@
         <version>0.4.0-SNAPSHOT</version>
     </parent>
 
-    <artifactId>kroxylicious-additional-filters</artifactId>
+    <groupId>io.kroxylicious.filter</groupId>
```

Comment:
Following [this stream of consciousness](https://github.com/kroxylicious/kroxylicious/pull/709#discussion_r1385872400) I'm going to revert the group of the filters to `io.kroxylicious` for the time being. Its easy enough to change it if we find a consensus on how to categorise things.

Reply:
@tombentley you left a reference to Maven group `io.kroxylicious.filter` in the kroxylicious-integration-test module.